src/test/java/org/folio/rest/unit/ItemDamagedStatusAPIUnitTest.java

```diff
@@ -0,0 +1,333 @@
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class ItemDamagedStatusAPIUnitTest {
+  private static final String DEFAULT_QUERY = "";
+  private static final int DEFAULT_OFFSET = 0;
+  private static final int DEFAULT_LIMIT = 10;
+  private static final String DEFAULT_LANGUAGE = "en";
+  private static final HashMap<String, String> DEFAULT_HEADERS = new HashMap<>();
+  private static final String INTERNAL_SERVER_ERROR_MSG = "Oops! Some error was happened on the server side.";
+  @Rule
+  public RunTestOnContext rule = new RunTestOnContext();
+  @Mock
+  private PostgresClientFactory pgClientFactory;
+  @Mock
+  private PostgresClient postgresClient;
+  @InjectMocks
+  private ItemDamagedStatusAPI itemDamagedStatusAPI = new ItemDamagedStatusAPI();
+
+
+  @Before
+  public void setUp(TestContext testContext) throws Exception {
+    MockitoAnnotations.initMocks(this);
+    when(pgClientFactory.getInstance(any(Context.class), any(Map.class))).thenReturn(postgresClient);
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnBadRequest(TestContext testContext) {
+    RuntimeException exception = new RuntimeException(INTERNAL_SERVER_ERROR_MSG);
+    doAnswer(setExceptionForHandlerArgument(6, exception))
+      .when(postgresClient)
+      .get(
+        eq(REFERENCE_TABLE),
+        eq(ItemDamageStatus.class),
+        eq(new String[]{"*"}),
+        any(CQLWrapper.class),
+        eq(true),
+        eq(true),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatuses(
+      DEFAULT_QUERY,
+      DEFAULT_OFFSET,
+      DEFAULT_LIMIT,
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnInternalServerError(TestContext testContext) {
+
+    doThrow(new RuntimeException(INTERNAL_SERVER_ERROR_MSG))
+      .when(postgresClient).get(
+      eq(REFERENCE_TABLE),
+      eq(ItemDamageStatus.class),
+      eq(new String[]{"*"}),
+      any(CQLWrapper.class),
+      eq(true),
+      eq(true),
+      any(Future.class)
+    );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatuses(
+      DEFAULT_QUERY,
+      DEFAULT_OFFSET,
+      DEFAULT_LIMIT,
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(500, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void postItemDamagedStatusesShouldReturnBadRequest(TestContext testContext) {
```

Comment:
The test name does not describe why a bad request should be returned. Under what circumstances should a bad request response be returned?

Reply:
Renamed